### PR TITLE
Issue #311 create cl make bin stage

### PR DESCRIPTION
**Pull request by marina-ricci**
Linked to #311. This issue creates a stage to bin the cluster catalogs in bins of richness and redshift and output a tomography TX file. Pipeline and config files are also written to run the CL pipeline, as well as the batch script for CCin2p3 and a notebook running the pipeline and exploring the input and output files.

### CODE REVIEW

**Comment by marina-ricci** [comment]
Pipeline and config files are also written to run the CL pipeline,

 as well as the batch script for CCin2p3 and

- notebook running the pipeline and exploring the input and output files : 
[Run_CL_pipeline](https://github.com/LSSTDESC/TXPipe/blob/issue_%23311_create_CL_make_bin_stage/notebooks/Run_CL_pipeline.ipynb
)

**Comment by marina-ricci** [comment]
@cavestruz : the PR is ready on my side. Normally you should have all info from the notebook! Please test the nersc launcher (just for the 1deg2 should be sufficient) as I did not test it. We should find better way to work at the two site with changing just the launcher file, but I did not find the proper way to do it. So you also have to change the first lines in the pipeline files.

File: examples/cosmodc2/pipeline-20deg2-CL-nersc.yml
Comment (cavestruz) [inline examples/cosmodc2/pipeline-20deg2-CL-nersc.yml:1]: - [ ] This has some lines that need updating to run, consistent with the 1deg2 version.

File: examples/cosmodc2/pipeline-20deg2-CL-in2p3.yml
Comment (cavestruz) [inline examples/cosmodc2/pipeline-20deg2-CL-in2p3.yml:1]: - [ ] This has some lines that need updating to run, consistent with the 1deg2 version.

File: notebooks/Run_CL_pipeline.ipynb
Comment (cavestruz) [inline notebooks/Run_CL_pipeline.ipynb:1]: I made some minor additions here to account for different in2p3 vs. nersc files.

File: examples/cosmodc2/config-20deg2-CL.yml
Comment (cavestruz) [inline examples/cosmodc2/config-20deg2-CL.yml:1]: Made minor change here, so there are separate yaml files depending on where the user is running the notebook.

**Review by cavestruz — CHANGES_REQUESTED** [review]
Primary changes are to change the 20sqdeg yaml files so this works.

**Comment by marina-ricci** [comment]
@cavestruz : I did some cleaning as there were typos in the files and reorganized the notebook to clarify the 2 approaches to run : direclty from jupyter or via pipeline/ceci. I commented all unused stages / input so that the pipeline chart for this PR is just:
image

File: examples/cosmodc2/config-20deg2-CL.yml
Comment (cavestruz) [reply examples/cosmodc2/config-20deg2-CL.yml:1]: Testing this out now with the `squeue` submission.  Will then check if latter portion of notebook runs on NERSC with the output of the 20deg2 submission that uses this yaml file.

**Review by cavestruz — APPROVED** [review]
Great - I am able to run everything in the notebook, and submit the `sbatch examples/cosmodc2/20deg2-nersc.sub` to run the tail end of cells in `Run_CL_counts_pipeline.ipynb`.

**Pull request merged**